CHANGELOG.md

```diff
@@ -6,6 +6,8 @@
 
 - Remove `pdata.AttributeMap.InitFromMap` (#4429)
 
+## 💡 Enhancements 💡
+- Support shell-like syntax for environment variable substitution defaults (`${ENV:-default}`) (#4458)
```

Comment:
Food for thought: We are going towards a direction of using "URI" like locations/identifiers for configs. I do understand that this is natural for lots of the people, but we should consider something like "env:ENV?default=foo" or in this direction.

Reply:
It feels unnatural to have `$ENV` when there is no default and `$env:ENV?default=foo` when there is a default; is that what we would be aiming for?

Reply:
I am proposing `${env:ENV}` and `{$env:ENV?default=foo}`

Reply:
That seems acceptable to me, it's close to PowerShell's syntax so it's not completely new. It will be a big breaking change, for end-users, though.

Reply:
I think for few versions (4-5) we will still support the old way. We are not going to just break the users :)

Reply:
Sounds like a sensible plan to me!